src/main/java/xyz/nucleoid/plasmid/util/WoodBlocks.java

```diff
@@ -0,0 +1,31 @@
+package xyz.nucleoid.plasmid.util;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+
+public class WoodBlocks {
+    public static BlockState saplingOf(BlockState state) {
+        Block block = state.getBlock();
+        if (block == Blocks.OAK_LEAVES) return Blocks.OAK_SAPLING.getDefaultState();
+        if (block == Blocks.BIRCH_LEAVES) return Blocks.BIRCH_SAPLING.getDefaultState();
+        if (block == Blocks.SPRUCE_LEAVES) return Blocks.SPRUCE_SAPLING.getDefaultState();
+        if (block == Blocks.JUNGLE_LEAVES) return Blocks.JUNGLE_SAPLING.getDefaultState();
+        if (block == Blocks.ACACIA_LEAVES) return Blocks.ACACIA_SAPLING.getDefaultState();
+        if (block == Blocks.DARK_OAK_LEAVES) return Blocks.DARK_OAK_SAPLING.getDefaultState();
+
+        return Blocks.AIR.getDefaultState();
```

Comment:
If this is going to be upstream code then IMO it should be held to greater standards. For instance, this could rather raise some kind of exception, or the function could return Optional<BlockState>